src/Common/src/System/IO/TempFileCollection.cs

```diff
@@ -5,11 +5,22 @@
 using System.Collections;
 using System.IO;
 
+// We keep this class in Common to allow utilizing it without taking a dependency on System.CodeDom
+#if CODEDOM
 namespace System.CodeDom.Compiler
+#else
+namespace System.IO.Internal
+#endif
```

Comment:
I know it isn't necessary, I only did it because I wanted to keep it in the IO folder so we can discover it a little better. Note that I'm going to be opening an issue about opening a Temp class of some sort in System.IO.